File: test_samples/java/dirty_function.java
// Dirty Java - contains unused methods

public class DirtyService {
    public int unusedHelper() {
        return 123;
    }

    public String formatData(String data) {
        return data.trim();
    }

    public static void main(String[] args) {
        DirtyService svc = new DirtyService();
        String val = svc.processValue("hello");
        System.out.println(svc.formatData(val));
    }

    public String processValue(String input) {
        return input.toUpperCase();
    }

    public int deepThought() {
        return 42;
    }

    public static int getConfig() {
        return 0;
    }
}

File: test_samples/typescript/dirty_function.ts
// Dirty TypeScript - contains unused functions

function helperUnused(): string {
    return "never called";
}

function formatData(data: string): string {
    return data.trim();
}

function main(): void {
    const val = processValue("hello");
    console.log(formatData(val));
}

function processValue(input: string): string {
    return input.toUpperCase();
}

function deepThought(): number {
    return 42;
}

export function getConfig(): object {
    return { debug: false };
}

export function setConfig(cfg: object): object {
    return cfg;
}

main();
